### PR TITLE
recharts: fix callback signatures, add missing props

### PR DESCRIPTION
**Pull request by Metavirulent**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://recharts.org/#/en-US/api callbacks all take different arguments, not just () => void
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/recharts/index.d.ts [hunk 1]
```diff
@@ -540,8 +545,9 @@ export interface ReferenceAreaProps {
 	viewBox?: ViewBox;
 	xAxis?: any;
 	yAxis?: any;
-	label?: string | number | React.ReactElement<any> | RechartsFunction;
+	label?: string | number | React.ReactElement<any> | ContentRenderer<ReferenceAreaProps>;
 	isFront?: boolean;
+    className?: string;
```
Comment (rapmue) [inline types/recharts/index.d.ts]: I'm not sure about this. But aren't `className`, `style` and other props part of `React.ReactElement` props? In my opinion, they should come from somewhere else e.g. extend another interface or class.

File: types/recharts/index.d.ts [hunk 2]
```diff
@@ -551,11 +557,13 @@ export interface ReferenceDotProps {
...
 	x: number | string;
 	y: number | string;
+	r: number;
 	alwaysShow?: boolean;
 	xAxis: any;
 	yAxis: any;
-	label?: string | number | React.ReactElement<any> | RechartsFunction;
+	label?: string | number | React.ReactElement<any> | ContentRenderer<ReferenceDotProps>;
 	isFront?: boolean;
+    className?: string;
```
Comment (rapmue) [inline types/recharts/index.d.ts]: dito

File: types/recharts/index.d.ts [hunk 3]
```diff
@@ -577,8 +585,9 @@ export interface ReferenceLineProps {
 	viewBox?: ViewBox;
 	xAxis?: any;
 	yAxis?: any;
-	label?: string | number | React.ReactElement<any> | RechartsFunction;
+	label?: string | number | React.ReactElement<any> | ContentRenderer<ReferenceLineProps>;
 	isFront?: boolean;
+    className?: string;
```
Comment (rapmue) [inline types/recharts/index.d.ts]: dito

File: types/recharts/index.d.ts [hunk 4]
```diff
@@ -589,7 +598,10 @@ export interface ResponsiveContainerProps {
 	height?: Percentage | number;
 	minWidth?: number;
 	minHeight?: number;
+    maxHeight?: Percentage | number;
 	debounce?: number;
+	id?: string | number;
+	className?: string;
```
Comment (rapmue) [inline types/recharts/index.d.ts]: dito

File: types/recharts/index.d.ts [hunk 5]
```diff
@@ -606,6 +618,7 @@ export interface ScatterProps extends Partial<CSSStyleDeclaration> {
 	isAnimationActive?: boolean;
 	animationBegin?: number;
 	animationEasing?: AnimationEasingType;
+    className?: string;
```
Comment (rapmue) [inline types/recharts/index.d.ts]: dito

File: types/recharts/index.d.ts [hunk 6]
```diff
@@ -659,6 +672,7 @@ export interface TextProps extends Partial<CSSStyleDeclaration> {
 	angle?: number;
 	textAnchor?: 'start' | 'middle' | 'end' | 'inherit';
 	verticalAnchor?: 'start' | 'middle' | 'end';
+	style?: React.CSSProperties;
```
Comment (rapmue) [inline types/recharts/index.d.ts]: dito

**Review by rapmue — CHANGES_REQUESTED** [review]
As noted in the comment, I'm not 100% sure if the properties `className` and `style` aren't inherited from `React.ReactElement`. It would be cleaner if they are inherited from somewhere else.

File: types/recharts/index.d.ts (hunk 1 above)
Comment (Metavirulent) [reply types/recharts/index.d.ts]: They are part of HTMLAttributes but recharts doesn't allow all HTMLAttributes. We can leave them out as they are not mentioned in the docs either.

Comment (rapmue) [reply types/recharts/index.d.ts]: yeah perfect. @Metavirulent please remove this one also (line 550) I think you missed this one in your latest commit.

**Review by rapmue — APPROVED** [review]
Looks good to me. Thanks for your contribution.

File: types/recharts/index.d.ts [hunk 7]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for Recharts 0.22
+// Type definitions for Recharts 0.23
```
Comment (royxue) [inline types/recharts/index.d.ts]: There is no version 0.23, you should either use 0.22 or 1.0 according to their websites

File: types/recharts/index.d.ts [hunk 8]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for Recharts 0.22
+// Type definitions for Recharts 1.0.0
```
Comment (rapmue) [inline types/recharts/index.d.ts]: It's not allowed to include a patchversion. i.e. only 1.0 should do the trick.

**Review by rapmue — CHANGES_REQUESTED** [review]

**Review by rapmue — APPROVED** [review]
Ok, should be fine now.

**Review by rapmue — APPROVED** [review]
I'm not sure if the bot noticed, that I approved already... Everything ok.

**Pull request merged**